pisek/task_jobs/program.py

```diff
@@ -91,6 +91,18 @@ def __init__(self, env: Env, name: str, **kwargs) -> None:
         self._program_pool: list[ProgramPoolItem] = []
         self._callback: Optional[Callable[[subprocess.Popen], None]] = None
 
+    @staticmethod
+    def _env_disjoint_union(
+        env1: dict[str, str], env2: dict[str, str]
+    ) -> dict[str, str]:
+        union = env1 | env2
+        if len(union) != len(env1) + len(env2):
+            duplicates = list(set(env1.keys()) & set(env2.keys()))
+            raise PipelineItemFailure(
+                "Duplicate environment variable: " + duplicates[0]
```

Comment:
I'm not sure this is the best error message. Wouldn't something like "ENV_VAR is used by the protocol and cannot be overwritten" be better?